sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -1197,6 +1197,35 @@ class UDFSuite extends QueryTest with SharedSparkSession {
       Row(Row(null)))
   }
 
+  test("SPARK-51402: Test TimeType in UDF") {
+    // Mocks
+    val mockTimeStr = "00:00:00.000000"
+    val input = Seq(java.time.LocalTime.parse(mockTimeStr)).toDF("currentTime")
```

Comment:
Could you import java.time.LocalTime above:
```scala
import java.time.{Instant, LocalDate, LocalTime}
```
This should improve test readability.